Modules/_decimal/_decimal.c

```diff
@@ -5699,9 +5697,29 @@ PyDec_GetConst(const PyObject *v)
     return MPD(v);
 }
 
+static void 
+free_api(void *capi)
+{
+    PyMem_Free(capi);
+}
+
+#define CAPSULE_NAME "_decimal._API"
+static void
+destroy_api(PyObject *obj)
+{
+    void *capi = PyCapsule_GetPointer(obj, CAPSULE_NAME);
+    free_api(capi);
+}
+
 static PyObject *
 init_api(void)
 {
+    void **_decimal_api = PyMem_Malloc(CPYTHON_DECIMAL_MAX_API * sizeof(void *));
```

Comment:
using `PyMem_Calloc(size, sizeof(void *))` in here would be better? I am not sure.

Reply:
Yes, it is definitely better. Thanks for pointing that out.